src/client/telemetry/localfileusersettings.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+
+import * as fs from 'fs-extra';
+import { getAppDataPath } from './appdata';
+import * as path from 'path';
+import { v4 } from 'uuid';
+
+interface IUserSettings {
+    readonly uniqueId: string;
+    readonly settingVersion: string;
+    readonly telemetryEnabled: boolean;
+}
+
+const settingVersionId = '1.0';
+const userSettingsFileName = 'usersettings.json';
+let settings: IUserSettings = {
+    uniqueId: v4(),
+    settingVersion: settingVersionId,
+    telemetryEnabled: true
+};
+
+// Application Path and User settings file name should be consistent with CLI
+export function readUserSettings(applicationPath?: string | undefined): IUserSettings {
+    if (!applicationPath) {
+        applicationPath = path.join(getAppDataPath(), 'Microsoft', 'PowerAppsCli');
+    }
+    const userSettingFilePath = path.join(applicationPath, userSettingsFileName);
+    if (fs.existsSync(userSettingFilePath)) {
+        const stream = fs.readFileSync(userSettingFilePath, 'utf8');
+        try {
+            settings = JSON.parse(stream) as IUserSettings;
+            if (settings.uniqueId === '') {
+                writeUserSettings(settings, applicationPath);
+            }
+        } catch (err) {
+            fs.unlinkSync(userSettingFilePath);
```

Comment:
nervous that this will throw if the file cannot be deleted... thus disabling this vscode extension for the full duration of the session.
Better to fail graceful by either using the async variant (which would turn this func into async, sigh...) or for now, wrap the unlinkSync into another try/catch and swallow and also skip writing the setting. Not having telemetry info for a VSCode session is the lesser evil than not having a functioning extension

Reply:
This file is copied from the PCF implementation in the pac cli repo. This is the kind of divergent change that doesn't scale when we should be using a robust shared implementation (i.e. module).

I'll add the try/catch though.

Reply:
and swallowing errors here is exactly the problem that I got the BatchedTelemetry code to stop doing so that we would actually be able to know that there are failures happening somewhere. In the end, we were able to fix the root cause of the exceptions in BT. that being said, the try/catch around this delete call is indeed in there. ;-)